packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/Abstractions/HttpRequestOptionsApi.cs

```diff
@@ -22,5 +22,6 @@ public HttpRequestOptionsApi(CSharpType type, ValueExpression original) : base(t
         public abstract HttpRequestOptionsApi ToExpression();
 
         public abstract CSharpType HttpRequestOptionsType { get; }
+        public abstract string ParameterName { get; }
```

Comment:
Also I think maybe it is better if we make this virtual and return `options`.
in this way Scm does not have to override this.
I say this is better because adding an abstract member is a breaking change.
I kind of think we do not have to have this abstract class - we could just have the Scm version here with everything virtual and azure could override those.

Reply:
> I kind of think we do not have to have this abstract class - we could just have the Scm version here with everything virtual and azure could override those.

If we have the SCM version implementation within an internal class, we hide the implementation from public, that's why I prefer to have abstract class with abstract methods to public only.

> But this still does not solve the issue: "when we found that we need something, we will have to go back here to add something"

I agree on this. I don't feel that abstraction a very flexible way to provide extension for sub-plugins, unless I have some misunderstanding about it.

Reply:
> But this still does not solve the issue: "when we found that we need something, we will have to go back here to add something"

This is a good point - new things would need to be virtual which would make the class inconsistent. Making everything virtual from the start with default implementations may be better. @m-nash thoughts? Either way we can iterate on this further as we get architect feedback on the design.

> I agree on this. I don't feel that abstraction is a very flexible way to provide extension for sub-plugins, unless I have some misunderstanding about it. My understanding is, when we find something needs to be abstracted, we come to MGC to replace it with abstraction.

We would come to MGC.CM plugin to replace it, assuming it belongs in the set of abstractions being created here.

Reply:
> Making everything virtual from the start with default implementations may be better.

Yes, this might be better. Now we are digging holes in the existing implementation to replace concrete with abstract wherever needed.